homeassistant/components/deconz/light.py

```diff
@@ -158,111 +170,139 @@ def color_mode(self) -> str:
         """Return the hs color value."""
-        return (self._device.hue / 65535 * 360, self._device.saturation / 255 * 100)
+        return (
+            int(self._device.hue) / 65535 * 360,
+            int(self._device.saturation) / 255 * 100,
+        )
 
     @property
-    def xy_color(self) -> tuple | None:
+    def xy_color(self) -> tuple[float, float] | None:
         """Return the XY color value."""
-        return self._device.xy
+        if self._device.xy is not None:
+            x_value: float = self._device.xy[0]
+            y_value: float = self._device.xy[1]
+            return (x_value, y_value)
+        return None
 
     @property
-    def is_on(self):
+    def is_on(self) -> bool:
         """Return true if light is on."""
-        return self._device.state
+        return bool(self._device.state)
 
-    async def async_turn_on(self, **kwargs):
+    async def async_turn_on(
+        self, **kwargs: bool | float | int | str | tuple[float, float]
+    ) -> None:
         """Turn on light."""
-        data = {"on": True}
+        data: dict[str, bool | float | int | str | tuple[float, float]] = {"on": True}
 
-        if ATTR_BRIGHTNESS in kwargs:
-            data["brightness"] = kwargs[ATTR_BRIGHTNESS]
+        if attr_brightness := kwargs.get(ATTR_BRIGHTNESS):
+            assert isinstance(attr_brightness, int)
+            data["brightness"] = attr_brightness
 
-        if ATTR_COLOR_TEMP in kwargs:
-            data["color_temperature"] = kwargs[ATTR_COLOR_TEMP]
+        if attr_color_temp := kwargs.get(ATTR_COLOR_TEMP):
+            assert isinstance(attr_color_temp, int)
+            data["color_temperature"] = attr_color_temp
 
-        if ATTR_HS_COLOR in kwargs:
+        if attr_hs_color := kwargs.get(ATTR_HS_COLOR):
+            assert isinstance(attr_hs_color, tuple)
             if COLOR_MODE_XY in self._attr_supported_color_modes:
-                data["xy"] = color_hs_to_xy(*kwargs[ATTR_HS_COLOR])
+                data["xy"] = color_hs_to_xy(*attr_hs_color)
             else:
-                data["hue"] = int(kwargs[ATTR_HS_COLOR][0] / 360 * 65535)
-                data["saturation"] = int(kwargs[ATTR_HS_COLOR][1] / 100 * 255)
+                data["hue"] = int(attr_hs_color[0] / 360 * 65535)
+                data["saturation"] = int(attr_hs_color[1] / 100 * 255)
 
         if ATTR_XY_COLOR in kwargs:
             data["xy"] = kwargs[ATTR_XY_COLOR]
 
-        if ATTR_TRANSITION in kwargs:
-            data["transition_time"] = int(kwargs[ATTR_TRANSITION] * 10)
+        if attr_transition := kwargs.get(ATTR_TRANSITION):
+            assert isinstance(attr_transition, float)
+            data["transition_time"] = int(attr_transition * 10)
         elif "IKEA" in self._device.manufacturer:
             data["transition_time"] = 0
 
-        if (alert := FLASH_TO_DECONZ.get(kwargs.get(ATTR_FLASH))) is not None:
+        if (
+            isinstance(flash := kwargs.get(ATTR_FLASH), str)
```

Comment:
What could this else be, besides an str?

Reply:
Changed it